pkg/splunk/client/azureblobclient.go

```diff
@@ -0,0 +1,238 @@
+}
+
+// InitAzureBlobClientSession initializes and returns a client session object
+func InitAzureBlobClientSession(ctx context.Context, appAzureBlobEndPoint string, storageAccountName string, secretAccessKey string) SplunkAzureBlobClient {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("InitAzureBlobClientSession")
+
+	scopedLog.Info("Initializing  AzureBlob Service for apps", "appAzureBlobEndPoint", appAzureBlobEndPoint)
+	var azureBlobClient SplunkAzureBlobClientTemp
+
+	// TODO target for next sprint CSPL XXXX List and download azure object
+	// This part need to be developed when the Azure blob rest handlers
+	// are built.
+
+	// var err error
+
+	// Enforcing minimum version TLS1.2
+	// tr := &http.Transport{
+	// 	TLSClientConfig: &tls.Config{
+	// 		MinVersion: tls.VersionTLS12,
+	// 	},
+	// }
+	// tr.ForceAttemptHTTP2 = true
+	// httpClient := http.Client{Transport: tr}
+
+	// config := &Config{
+	// 	MaxRetries: 3,
+	// 	HTTPClient: httpClient,
+	// }
+
+	// azureBlobClient, err = NewAzureSession(ctx, appAzureBlobEndPoint, storageAccountName, secretAccessKey, config)
+	// if err != nil {
+	// 	scopedLog.Info("Error creating new AzureBlob Client Session", "err", err)
+	// 	return nil
+	// }
+
+	azureBlobClient = SplunkAzureBlobClientTemp{client: http.Client{}}
+	return &azureBlobClient
+
+}
+
+// GetAppsList get the list of apps from remote storage
+func (client *AzureBlobClient) GetAppsList(ctx context.Context) (RemoteDataListResponse, error) {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("GetAppsList")
+
+	scopedLog.Info("Getting Apps list", " S3 Bucket", client.BucketName, "Prefix", client.Prefix)
+
+	azureBlobClient := client.Client
+
+	// Create a bucket list command for all files in bucket
+
+	//TODO: revise it in next sprint. had to use map temporary to let build work
+	opts := make(map[string]string)
+
+	opts["Prefix"] = client.Prefix
+	opts["MaxEnteries"] = "4000"
+
+	// List all objects from a bucket-name with a matching prefix.
+	listResponse, err := azureBlobClient.ListApps(context.Background(), client.BucketName, opts)
+
+	//TODO actual working logic to be covered in CSPL XXXX list azure apps
+	remoteDataListResponse, err := extractResponse(listResponse)
+
+	if err != nil {
+		scopedLog.Error(err, "Failed to get list apps response", "Azure Blob Bucket", client.BucketName)
+		return remoteDataListResponse, err
+	}
+
+	return remoteDataListResponse, err
+}
+
+func extractResponse(listBlobsXML []byte) (RemoteDataListResponse, error) {
+	remoteDataListResponse := RemoteDataListResponse{}
+
+	err := json.Unmarshal(listBlobsXML, &(remoteDataListResponse.Objects))
+	if err != nil {
+		return remoteDataListResponse, err
+	}
+
+	return remoteDataListResponse, nil
+}
+
+// DownloadApp downloads an app package from remote storage
+func (client *AzureBlobClient) DownloadApp(ctx context.Context, downloadRequest RemoteDataDownloadRequest) (bool, error) {
```

Comment:
I am unsure why we need 2 return values here. err itself should be suffice right ?

Reply:
Good point. I used the same approach as other clients like AWSS3Client :) Happy to look into it in one of the future sprints. There are some places the callers are using the bool return value also which seems to be redundant. Legacy code.